main.py

```diff
@@ -0,0 +1,44 @@
+import os
+import discord
+from discord.app_commands import command
+from dotenv import load_dotenv
+from discord.ext import commands
+import asyncio
+
+def main():
+    load_dotenv()
+
+    TOKEN = os.getenv('DISCORD_TOKEN')
+    GUILD = os.getenv('DISCORD_GUILD')
```

Comment:
Here I would make two changes
1. Create a function to handle env variables
2. This function returns a structured data, with Discord info.

SMTH like
```suggestion
    discord_config = get_discord_config()

```

Then you would have a function

```python
def setup_discord_config():
    load_dotenv()

    TOKEN = os.getenv('DISCORD_TOKEN')
    if TOKEN is None:
       raise Exception ("Missing DISCORD_TOKEN variable")
    GUILD = os.getenv('DISCORD_GUILD')
    if GUILD is None:
       raise Exception ("Missing DISCORD_GUILD variable")

   return dict(token=TOKEN,guild=GUILD)
```

BTW, I would avoid using a raw dict, so check it out [typed dict](https://peps.python.org/pep-0589/).

Reply:
Basically, here
1. If missing a token, you already raises an error - 
Doesn't look to make sense to keep going, without a token.

This could be related to the Fail Fast principle ([martil fowler paper](https://martinfowler.com/ieeeSoftware/failFast.pdf),[ medium article])(https://medium.com/@christian.ppl/introduction-to-the-fail-fast-principle-in-software-development-865ccab28979)

> In [systems design](https://en.wikipedia.org/wiki/Systems_design), a fail-fast system is one that immediately reports at its interface any condition that is likely to indicate a failure. Fail-fast systems are usually designed to stop normal operation rather than attempt to continue a possibly flawed process. Such designs often check the system's state at several points in an operation, so that any failures can be detected early. The responsibility of a fail-fast module is detecting errors, and then letting the next-highest level of the system handle them.
https://en.wikipedia.org/wiki/Fail-fast_system

Reply:
2 - Then you simplify your main function. Now it doesn't need to care about os.variable, it's more readable something like:


1. setup()
3. run()

than

1. load A
2. get var B
3. run (A,B)